.github/workflows/typescript.yaml

```diff
@@ -130,6 +130,113 @@ jobs:
           cd ../sdk
           VITE_IOTA_BIN="iota" pnpm test:e2e
 
+  dapp_e2e_test:
+    name: Run dapp E2E Tests
+    runs-on: ubuntu-latest
+    env:
+      NEXT_PUBLIC_NAMES_DAPP_DEFAULT_NETWORK: localnet
+      NEXT_PUBLIC_NAMES_DAPP_CONFIG: |
+        {
+          "localnet": {
+            "network": "localnet",
+            "namesDisplayUrl": "http://localhost",
+            "indexerUrl": "http://localhost",
+            "baseUrl": "http://localhost:3000"
+          }
+        }
+    services:
+      postgres:
+        image: postgres:15
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: postgrespw
+          POSTGRES_DB: iota_indexer
+          POSTGRES_HOST_AUTH_METHOD: trust
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+          - 5432:5432
+    defaults:
+      run:
+        working-directory: ./dapp
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          fetch-depth: 1
+
+      - uses: actions/setup-node@cdca7365b2dadb8aad0a33bc7601856ffabcc48e # v4.3.0
+        with:
+          node-version: 20
+
+      - name: Install Homebrew
+        run: |
+          /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+          echo "/home/linuxbrew/.linuxbrew/bin" >> $GITHUB_PATH
+
+      - name: Install IOTA using Homebrew
+        run: brew install iotaledger/tap/iota
+
+      - name: Do a global PNPM install
+        run: |
+          npm install -g pnpm
+
+      - name: Install Dependencies
+        run: |
+          pnpm install
+        working-directory: ./
+
+      - name: Build Dapp
+        run: pnpm dapp build
+        working-directory: ./
+
+      - name: Download IOTA Wallet Extension
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          pnpm run test:prepare
+
+      - name: Install Playwright Browsers
+        run: |
+          pnpm run playwright:install
+
+      - name: Prepare local network & Run Dapp e2e tests
+        working-directory: ./
+        run: |
+          iota start --with-faucet --faucet-amount 100000000000000 --with-indexer --with-graphql --force-regenesis --epoch-duration-ms 120000 > /dev/null 2>&1 &
+          sleep 10
+
+          iota client --yes new-env --alias localnet --rpc http://127.0.0.1:9000 --graphql http://127.0.0.1:9125
+          iota client switch --env localnet
+          iota client faucet
+          sleep 5
+
+          cd scripts
+          export NETWORK=localnet
+          pnpm ts-node init/init.ts localnet
+          pnpm ts-node tests/register-name.ts
+          sleep 5
+
+          cat ./package-info/localnet.json
+
+          sed -i "/REPLACE PLACEHOLDER/i\\    localnet: $(jq -c . ./package-info/localnet.json)," ../sdk/src/constants.ts
```

Comment:
is it possible to create a `package.json` script that does this automatically? would be nice for local testing